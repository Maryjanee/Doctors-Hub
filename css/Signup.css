body {
  background: #ffb400;
}
/*# sourceMappingURL=Signup.css.map */