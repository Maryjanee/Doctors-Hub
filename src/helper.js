/* eslint-disable import/prefer-default-export */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable arrow-body-style */
import { BrowserRouter } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import rootReducer from './reducers';

export const renderComponent = (Component, props = {}) => {
  return (
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  );
};

export const renderWithRedux = (component,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  } = {}) => ({
  ...render(
    <Provider store={store}>
      renderComponent(component)
    </Provider>,
  ),
});
