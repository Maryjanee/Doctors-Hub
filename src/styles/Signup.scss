$yellow: #ffb400;

body {
  background: $yellow;
}
