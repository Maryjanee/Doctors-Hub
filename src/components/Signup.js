import { Link, Redirect, useHistory } from 'react-router-dom';
import { useState } from 'react';
import AuthService from '../services/auth-service';
import '../styles/Signup.scss';

const Signup = () => {
  const history = useHistory();
  const [SignupError, setSignupError] = useState(null);
  const [signupForm, updateSignupForm] = useState({
    username: '',
    email: '',
    password: '',
  });

  const handleChange = event => {
    const { name, value } = event.target;
    updateSignupForm(prev => ({
      ...prev,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();
    console.log({ signupForm });
    const { username, email, password } = signupForm;
    if (username && email && password) {
      const response = await AuthService.signUp(signupForm);
      console.log(response);
      if (response && !response.error) {
        history.push('/');
      }
      if (!response || response.error) {
        setSignupError('An Error Occurred. Please try again');
      }
    }
  };

  if (localStorage.getItem('auth-token')) {
    return <Redirect to="/doctors" />;
  }

  return (
    <div className="main-bg">
      {SignupError && <div className="notification-bar">{SignupError}</div>}
      <div className="container">
        <h1 className="logo">Doctors Hub</h1>

        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <input
              type="text"
              id="username"
              name="username"
              onChange={handleChange}
              placeholder="Username"
            />

            <input
              type="email"
              id="email"
              name="email"
              onChange={handleChange}
              placeholder="Email"
            />

            <input
              type="password"
              id="password"
              name="password"
              onChange={handleChange}
              placeholder="Password"
            />

            <button type="submit" className="submit">
              Create an Account
            </button>
          </form>
          <p className="text-center small">
            Already a member?
            <Link to="/" className="create">
              Log In
            </Link>
          </p>
        </div>
      </div>
    </div>

  );
};
export default Signup;
