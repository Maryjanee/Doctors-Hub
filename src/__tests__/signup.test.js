import renderer from 'react-test-renderer';
import Signup from '../components/Signup';
import { renderComponent } from '../helper';

describe('Login Page', () => {
  it('deep snapshot renders correctly', () => {
    const component = renderComponent(Signup);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
