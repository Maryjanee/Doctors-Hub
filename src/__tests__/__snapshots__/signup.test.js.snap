// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page deep snapshot renders correctly 1`] = `
<div
  className="main-bg"
>
  <div
    className="container"
  >
    <h1
      className="logo"
    >
      Doctors Hub
    </h1>
    <div
      className="form-container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <button
          className="submit"
          type="submit"
        >
          Create an Account
        </button>
      </form>
      <p
        className="text-center small"
      >
        Already a member?
        <a
          className="create"
          href="/"
          onClick={[Function]}
        >
          Log In
        </a>
      </p>
    </div>
  </div>
</div>
`;
